src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Tests.cs

```diff
@@ -253,5 +255,57 @@ public void SkipRecursionIntoBaseDirectorySymlink()
 
             Assert.Null(reader.GetNextEntry());
         }
+
+        [ConditionalTheory]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        [MemberData(nameof(GetPaxAndGnuTestCaseNames))]
+        public void ResultingArchive_CanBeExtractedByOtherTools(string testCaseName)
+        {
+            if (testCaseName == "specialfiles" && !PlatformDetection.IsSuperUser)
+            {
+                throw new SkipTestException("specialfiles needs sudo permissions for extraction");
+            }
+
+            using TempDirectory root = new TempDirectory();
+
+            string archivePath = GetTarFilePath(CompressionMethod.Uncompressed, TestTarFormat.gnu.ToString(), testCaseName);
+            string tarExtractedPath = Path.Join(root.Path, "tarExtracted");
+            Directory.CreateDirectory(tarExtractedPath); // rename to gnuTarExtractedPath et al
+
+            // extract with /usr/bin/tar
+            ExtractWithTarTool(archivePath, tarExtractedPath);
+
+            // create archive with TarFile
+            string dotnetTarArchive = Path.Join(root.Path, "dotnetTarArchive");
+            TarFile.CreateFromDirectory(tarExtractedPath, dotnetTarArchive, includeBaseDirectory: false);
+
+            // extract TarFile's archive with /usr/bin/tar
+            string dotnetTarExtractedPath = Path.Join(root.Path, "dotnetTarExtracted");
+            Directory.CreateDirectory(dotnetTarExtractedPath);
+            ExtractWithTarTool(dotnetTarArchive, dotnetTarExtractedPath);
+
+            // compare results
+            Dictionary<string, FileSystemInfo> expectedEntries = GetFileSystemInfosRecursive(tarExtractedPath).ToDictionary(fsi => fsi.FullName.Substring(tarExtractedPath.Length));
+
+            foreach (var actualEntry in GetFileSystemInfosRecursive(dotnetTarExtractedPath))
+            {
+                string keyPath = actualEntry.FullName.Substring(dotnetTarExtractedPath.Length);
+
+                Assert.True(expectedEntries.TryGetValue(keyPath, out FileSystemInfo expectedEntry), "Entry not expected.");
+                Assert.Equal(expectedEntry.Attributes, actualEntry.Attributes);
+                Assert.Equal(expectedEntry.LinkTarget, actualEntry.LinkTarget);
```

Comment:
Did the link tests pass this check? Weren't the `LinkTarget` paths full paths? If they were, then these need the prefix folder truncated too, just like in lines 292 and 288.

Reply:
LinkTarget is not converted to full path, it is what the user wrote.

Reply:
Is that the same case with the expected items?

Reply:
Well, the expected items are also wrapped by a FileSystemInfo, so yes.